app/src/main/java/org/oppia/app/topic/ViewPagerAdapter.kt

```diff
@@ -0,0 +1,47 @@
+package org.oppia.app.topic
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import org.oppia.app.topic.overview.TopicOverviewFragment
+import org.oppia.app.topic.play.TopicPlayFragment
+import org.oppia.app.topic.review.TopicReviewFragment
+import org.oppia.app.topic.train.TopicTrainFragment
+
+class ViewPagerAdapter(fm: FragmentManager, private val numOfTabs: Int, private val topicId: String) :
+  FragmentStatePagerAdapter(fm) {
+
+  override fun getItem(position: Int): Fragment? {
+    val args = Bundle()
+    args.putString(TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY, topicId)
+    when (position) {
```

Comment:
Suggest using an enum to have type safety with each tab, and automatically tie them to a total count. Consider something like:

```kotlin
override fun getItem(position: Int): Fragment? {
  val args = ...
  return when (TopicTab.getTabForPosition(position)) {
    TopicTab.OVERVIEW -> {
      val topicOverviewTab = TopicOverviewFragment()
      tabOverviewTab.arguments = args
      topicOverviewTab
    }
    TopicTab.PLAY -> { ... }
    TopicTab.TRAIN -> { ... }
    TopicTab.REVIEW -> { ... }
  }
}

private enum class TopicTab(val position: Int) {
  OVERVIEW(position = 0),
  PLAY(position = 1),
  TRAIN(position = 2),
  REVIEW(position = 3)

  companion object {
    fun getTabForPosition(position: Int): TopicTab {
      val ordinal = checkNotNull(TopicTab.values.map(TopicTab::getPosition).find(position)) {
        "No tab corresponding to position: $position"
      }
      return TopicTab.values[ordinal]
    }
  }
}
```

Note that no ``else`` case needs to be handled here since all of the tabs are handled in the ``when`` statement, and the static method ensures the position integer is always correctly mapped to a tab enum constant.

Reply:
Resolved